test/Icons.Test/Models/IconHttpRequestTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Net;
+using Bit.Icons.Models;
+using Bit.Icons.Services;
+using Bit.Test.Common.MockedHttpClient;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Net.Http.Headers;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Icons.Test.Models;
+
+public class IconHttpRequestTests
+{
+    [Fact]
+    public async Task FetchAsync_FollowsTwoRedirectsAsync()
+    {
+        var handler = new MockedHttpMessageHandler();
+
+        var request = handler
+            .Fallback
+            .WithStatusCode(HttpStatusCode.Redirect)
+            .WithContent("text/html", "<html><head><title>Redirect 2</title></head><body><a href=\"https://test.local\">Redirect 3</a></body></html>")
+            .WithHeader(HeaderNames.Location, "https://test.local");
```

Comment:
I actually learned about special TLD's that are meant for testing/examples that I think we should use for URI's that we don't need to actually be real. https://www.rfc-editor.org/rfc/rfc2606.html#section-2

Reply:
ah, I had actually thought `.local` was one...